Document custom header ids for nextra.site (#1706)
shuding/nextra: docs/pages/docs/guide/markdown.mdx

Visit https://nextjs.org.
```

> ### Custom Header Ids
> 
> You can specify custom header ids using the format `## Header [#custom-id]`. For example:
> 
> ```md filename="Markdown"
> ## Long header about Nextra [#about-nextra]
> ```
> 
> In this example, `#about-nextra` will be used as the header link, replacing the default `#long-header-about-nextra`.

Reviewer: ````suggestion
### Custom Heading Id

You can specify custom heading id using the format `## My heading [#custom-id]`. For example:

```md filename="Markdown"
## Long heading about Nextra [#about-nextra]
```

In this example, `#about-nextra` will be used as the heading link, replacing the default `#long-heading-about-nextra`.
````
